src/core/Carousel/Carousel.tsx

```diff
@@ -0,0 +1,298 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import React from 'react';
+import cx from 'classnames';
+import { useTheme } from '../utils';
+import { IconButton } from '../Buttons';
+import SvgChevronLeft from '@itwin/itwinui-icons-react/cjs/icons/ChevronLeft';
+import SvgChevronRight from '@itwin/itwinui-icons-react/cjs/icons/ChevronRight';
+import '@itwin/itwinui-css/css/carousel.css';
+
+export type CarouselProps = {
+  currentIndex?: number;
+  autoPlay?: boolean;
+} & React.ComponentPropsWithoutRef<'div'>;
+
+const CarouselContext = React.createContext<
+  | {
+      currentIndex: number;
+      setCurrentIndex: (index: number | ((old: number) => void)) => void;
+      slideCount: number;
+      setSlideCount: (length: number | ((old: number) => void)) => void;
+      keysPressed: Record<string, boolean>;
+    }
+  | undefined
+>(undefined);
+
+export const Carousel = (props: CarouselProps) => {
+  const { currentIndex: userCurrentIndex = 0, children, ...rest } = props;
+
+  useTheme();
+
+  const [currentIndex, setCurrentIndex] = React.useState(userCurrentIndex);
+  const [slideCount, setSlideCount] = React.useState(0);
+
+  const [keysPressed, setKeysPressed] = React.useState<Record<string, boolean>>(
+    {},
+  );
+
+  const handleKeyDown = (event: React.KeyboardEvent) => {
+    if (event.altKey || event.ctrlKey || event.metaKey) {
+      return;
+    }
+
+    if (event.key === 'ArrowLeft' || event.key === 'ArrowRight') {
+      setKeysPressed((old) => ({ ...old, [event.key]: true }));
+    }
+
+    switch (event.key) {
+      case 'ArrowLeft': {
+        setCurrentIndex((old) => (slideCount + old - 1) % slideCount);
```

Comment:
Why not just `old + 1`?

Reply:
You mean `old - 1`?

I'm using `%` to loop over from one edge to another.

Reply:
Oh, ok. Then why `(old - 1) % slideCount` is not enough?

Reply:
you tell me 😄 
![image](https://user-images.githubusercontent.com/9084735/156370920-c956b820-2c21-4675-9a13-286d971bade7.png)

Reply:
It is expected to always loop over? Wont be situations, that you want to stop it at the edge?

Reply:
yeah..on quick googling couldnt find anything on this, so.. I guess doesnt matter